src/Httpfs/FileManager.java

```diff
@@ -0,0 +1,92 @@
+package Httpfs;
+
+import java.io.*;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class FileManager {
+    Path directory;
+
+
+
+    FileManager(Path directory) {
+        this.directory = directory;
+    }
+
+    /**
+     * Ensure path is within directory
+     */
+    public boolean isFilePathValid(String path){
```

Comment:
This seems good. But it's never called from writeToFile or readFile.

Reply:
I see the check for /../ when parsing the url path, so any url with that won't even make it to this point. Left it in incase this changes.